amf-shapes/shared/src/main/scala/amf/shapes/internal/annotations/Annotations.scala

```diff
@@ -15,6 +15,16 @@ object ParsedJSONSchema extends AnnotationGraphLoader {
     Some(ParsedJSONSchema(value))
 }
 
+case class DeclarationsKey(key: String) extends EternalSerializedAnnotation {
```

Comment:
You introduced another annotation so that it can "survive" the json-ld right? I would change the name to avoid clashing with the one used by ALS.

The other way is to remove this one and instead make the one used by ALS an eternal and make the value by the entry key. This way als can still use it and we can dump and parse it from json-ld